web/src/app/services/data-store/data-store.service.ts

```diff
@@ -140,7 +141,17 @@ export class DataStoreService {
     userEmail: string,
     userId: string
   ): Observable<List<Survey>> {
-    return this.db
+    const surveyChangeAction = (surveys: DocumentChangeAction<unknown>[]) =>
```

Comment:
Let's extract this into its own function with clear TSDoc and name as to what this does (eg `documentChangeToSurvey()`)